qemu/tests/qemu_guest_agent.py

```diff
@@ -2530,15 +2530,61 @@ def gagent_check_fsinfo(self, test, params, env):
         mountpoint,disk's name and serial number.
 
         steps:
-        1) check file system type of every mount point.
-        2) check disk name.
-        3) check disk's serial number.
+        1) Check filesystem usage statistics
+        2) check file system type of every mount point.
+        3) check disk name.
+        4) check disk's serial number.
 
         :param test: kvm test object
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
 
         """
+        def qga_guest_diskusage(mountpoint):
+            """
+            Send cmd in guest to get disk usage.
+            :param mountpoint: the mountpoint of filesystem
+            """
+            cmd_get_diskusage = params["cmd_get_disk_usage"] % mountpoint
+            disk_usage_guest = session.cmd(cmd_get_diskusage).strip().split()
+            disk_total_guest = int(disk_usage_guest[0])
+            if params["os_type"] == "windows":
+                # Just can get total and freespace disk usage from windows.
+                disk_freespace_guest = int(disk_usage_guest[1])
+                disk_used_guest = int(disk_total_guest - disk_freespace_guest)
+            else:
+                disk_used_guest = int(disk_usage_guest[1])
+            disk_total_qga = int(fs["total-bytes"])
+            disk_used_qga = int(fs["used-bytes"])
+            diff_total_qga_guest = abs(disk_total_guest - disk_total_qga)
+            diff_used_qga_guest = abs(disk_used_guest - disk_used_qga)
+            return (diff_total_qga_guest, diff_used_qga_guest)
+
+        def check_usage_qga_guest(mount_point):
+            """
+            Contrast disk usage from guest and qga that needed
+            to call previous function 'qga_guest_diskusage'.
+            :param mountpoint: the mountpoint of filesystem
+            """
+            disk_usage_guest = qga_guest_diskusage(mount_point)
+            diff_total_qgaguest = int(disk_usage_guest[0])
+            diff_used_qgaguest = int(disk_usage_guest[1])
+            if (diff_total_qgaguest != 0 or
+                    diff_used_qgaguest != 0):
+                if mount_pt == 'C:' or mount_pt == '/':
+                    # Disk 'C:' and '/' usage have a floating interval,
+                    # so set a safe value '10485760'.
+                    if (diff_total_qgaguest > 10485760 or
+                            diff_used_qgaguest > 10485760):
```

Comment:
Would you explain why the safe value is `10485760` here? Please notice that other users might set different disk size and there disks might have different usage condition.

Reply:
Also, Windows should have different from Linux for the root disk general condition.

Reply:
The safe value just for floating space of two queries that checking 'C:' or '/', disk usage. because the disk usage of them may be different from the previous second, but it won't be a big gap certainly,so this value I just test 100 times and estimated value.

> Would you explain why the safe value is `10485760` here? Please notice that other users might set different disk size and there disks might have different usage condition.

The safe value just for floating space of two queries that checking 'C:' or '/', disk usage. because the disk usage of them may be different from the previous second, but it won't be a big gap certainly,so this value I just test 100 times and estimated value.

Reply:
@xiagao Would you help confirm this? If you guys all think this value is okay, then it's okay for me, thanks.

Reply:
Actually the total disk space should have the fixed value, demeng need to check it first.
And about the diff value, the time to get the disk size twice is relatively short and no big data written, so there is no big different, 10485760 is ok for me.